File: kappuccino-java/src/main/java/com/heitorcolangelo/kappuccino/java/assertion/VisibilityAssertion.java
package com.heitorcolangelo.kappuccino.java.assertion;

import com.heitorcolangelo.kappuccino.core.ViewMatcherMethod;
import com.heitorcolangelo.kappuccino.core.assertion.BaseVisibilityAssertion;

import javax.annotation.ParametersAreNonnullByDefault;

@ParametersAreNonnullByDefault
public class VisibilityAssertion extends BaseVisibilityAssertion {

    private VisibilityAssertion(ViewMatcherMethod... viewMatcherMethods) {
        super(viewMatcherMethods);
    }

    public static void displayed(ViewMatcherMethod... viewMatchers) {
        VisibilityAssertion assertion = new VisibilityAssertion(viewMatchers);
        assertion.setAssertion(true);
        assertion.doAssert();
    }

    public static void notDisplayed(ViewMatcherMethod... viewMatchers) {
        VisibilityAssertion assertion = new VisibilityAssertion(viewMatchers);
        assertion.setAssertion(false);
        assertion.doAssert();
    }

}

File: kappuccino-kotlin/src/main/kotlin/com/heitorcolangelo/kappuccino/kotlin/VisibilityAssertion.kt
package com.heitorcolangelo.kappuccino.kotlin

import com.heitorcolangelo.kappuccino.core.ViewMatcherMethods
import com.heitorcolangelo.kappuccino.core.VisibilityAssertion

private val visibilityAssertion = VisibilityAssertion()
private val viewMatcherMethods = ViewMatcherMethods()

fun displayed2(matcher: VisibilityAssertion.Displayed.() -> Unit) {
    VisibilityAssertion.Displayed(visibilityAssertion, viewMatcherMethods).apply { matcher() }
}

fun notDisplayed2(matcher: VisibilityAssertion.NotDisplayed.() -> Unit) {
    VisibilityAssertion.NotDisplayed(visibilityAssertion, viewMatcherMethods).apply { matcher() }
}

